llama2.mojo

```diff
@@ -135,7 +135,7 @@ struct Matrix:
 
 fn read_val_int(inout buf: FileBuf) raises -> Int:
     # DTypePointer[DType.ui8](buf.data).bitcast[DType.ui8]()
-    let data = buf.data.offset(buf.get_offset()).bitcast[DType.uint32]()
+    let data = buf.data.offset(buf.get_offset()).bitcast[DType.int32]()
```

Comment:
Why the type changing? Are there negative values in the tokenizer?

Reply:
The vocab_size maybe negative to note unshared weights